### ISSUE TITLE
Explore: Filter disappear when adding a label matcher via the UI

### ISSUE THREAD

**Issue by cyriltovena** [grafana/grafana#19512]
**What would you like to be added**:

Currently in Explore with Loki datasource when you have a multiple filter expressions:

```
{app="foo"}  != "not that errror" != "no that one too" != "neither this one"...
```

And you click on the `+` button to add a new label matcher from a label value, it will add the label matcher but also remove all filter ending up with `{app="foo",new="matcher"}`.

**Why is this needed**:

It would be better if we don't have to rewrite filter expressions from scratch when adding a label matcher from the UI.

**Comment by hugohaggmark** [comment]
Thanks @cyriltovena

**Comment by ivanahuckova** [comment]
Issue fixed in #20570.
